gitops/yaml/resources.go

```diff
@@ -82,3 +85,30 @@ func AddKustomize(fs afero.Fs, name string, items []string, path string) error {
 	content = append(content, map[string]interface{}{name: items})
 	return MarshalItemToFile(fs, path, content)
 }
+
+// MergeKustomizeResource rewrites existing kustomization.yaml by adding specified items under resources section
+func MergeResourcesIntoKustomize(fs afero.Fs, name string, res ...string) error {
```

Comment:
Can you add a test for this function?

https://github.com/redhat-appstudio/application-service/blob/main/gitops/yaml/resources_test.go should provide some pointers on how we unit tested some of the fs operations with afero (including read only fs)